server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -516,8 +515,8 @@ private boolean isLocalOrSnapshotRecovery() {
         // In this case when the primary mode is false, we need to upload segments to Remote Store
         // This is required in case of snapshots/shrink/ split/clone where we need to durable persist
         // all segments to remote before completing the recovery to ensure durability.
-
         return (indexShard.state() == IndexShardState.RECOVERING && indexShard.shardRouting.primary())
+            && indexShard.recoveryState() != null
```

Comment:
`recoveryState` is never null . It is set to `DONE` after the recovery is completed  .

Reply:
The recoveryState is nullable and is not set in the ctor as well. There is a possibility that recoveryState can be null.

https://github.com/opensearch-project/OpenSearch/blob/74b2d7daf4f5caf42d71724b2dcb745a31e6a884/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L297-L298

https://github.com/opensearch-project/OpenSearch/blob/74b2d7daf4f5caf42d71724b2dcb745a31e6a884/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L2539-L2546

Let me know if you still think it can never be null.